codegen/types.go

```diff
@@ -132,8 +132,8 @@ func GoTypeDef(dt design.DataType) string {
 		return fmt.Sprintf("map[%s]%s", keyDef, elemDef)
 	case design.Object:
 		return goTypeDefObject(actual)
-	case *design.UserTypeExpr:
-		return GoTypeName(actual)
+	case design.UserType:
+		return GoTypeName(actual.Attribute().Type)
```

Comment:
I think the test isn't correct. `GoTypeDef` should return the type name for user types.  Also `GoTypeName` should return the `TypeName` field of the embedded `UserTypeExpr` for `MediaTypeExpr`.